web/src/user_profile.ts

```diff
@@ -1085,12 +1099,19 @@ function toggle_submit_button($edit_form: JQuery): void {
     const current_values = get_current_values($edit_form);
     const $submit_button = $("#user-profile-modal .dialog_submit_button");
     const full_name_value = $edit_form.find<HTMLInputElement>("input[name='full_name']").val()!;
+    const $short_name = $edit_form.find<HTMLInputElement>("input[name='bot_short_name']");
+    const short_name_value = $short_name.length > 0 ? $short_name.val()!.trim() : "";
 
     if (full_name_value.trim() === "") {
         $submit_button.prop("disabled", true);
         return;
     }
 
+    if ($short_name.length > 0 && short_name_value === "") {
```

Comment:
I still don't get why we need to check `.length` here. Is there some case where input is not visible or something like that?